On NCBI: https://www.ncbi.nlm.nih.gov/data-hub/genome/GCF_018345385.1/
and easily downloadable: https://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/494514/100/GCF_018345385.1_ASM1834538v1/ (gff and gtf files for [[SnpEff]] database building)

Downloaded from website (clicking): https://www.ncbi.nlm.nih.gov/data-hub/genome/?taxon=494514
Move to dingo and then used scp to move to farm
```
# on dingo
cd /mnt/amazon/MECU/catetemp/
unzip ncbi_dataset.zip 
scp -r ncbi_dataset/data/* cbquinn@farm.cse.ucdavis.edu:/group/ctbrowngrp2/cbquinn/fox4/ref/

# on farm
cd /group/ctbrowngrp2/cbquinn/fox4/ref/
unzip ncbi_dataset.zip

cd /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/
# index reference for bwa
bwa index GCF_018345385.1_ASM1834538v1_genomic.fna
# index reference genome with samtools
samtools faidx GCF_018345385.1_ASM1834538v1_genomic.fna

# index reference for GATK3.6
module load GATK/3.6
module load picardtools/2.7.1
cd /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/
REF=GCF_018345385.1_ASM1834538v1_genomic.fna
java -jar ${PICARD}/picard.jar CreateSequenceDictionary \
R=$REF \
O=${REF%.fna}.dict

# make bedfiles of autosomes
# remember samtools is 1-based and bedtools is 0-based
sed -n 1,24p GCF_018345385.1_ASM1834538v1_genomic.fna.fai | awk -F "\t" '{ print $1, 0, $2-1 }' > /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.bed

# get sum of autosomes
awk '{ sum+=$3} END {print sum}' autosomes.bed > autosomes.total

# get sum of mappable autosomes

awk '{$4=$3-$2}1' /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/genmap/artic/GCF_018345385.1_ASM1834538v1_genomic_E2_high_mapability_mask.bed | awk '{ sum+=$4} END {print sum}' > /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/genmap/artic/GCF_018345385.1_ASM1834538v1_genomic_E2_high_mapability_mask.total

# make 1 MB windows bedfile of autosomes
bedtools makewindows -b /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.bed -w 1000000 -s 1000000 > /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes_1MBwindows.bed

# get bed file of CDS
# note! gff/gtf are 1-based
awk '$3 == "CDS"' /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/genomic.gff | awk -v OFS=$'\t' '{print $1, $4-1, $5}' > /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/CDS.bed

# sort and compress GTF/GFF
cd /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/
grep -v "#" genomic.gff | sort -k1,1 -k4,4n -k5,5n -t$'\t' | bgzip -c > genomic.gff.gz
# index GTF/GFF 
tabix -p gff genomic.gff.gz

# get fasta of CDS in arctic fox (for fidnign gerp in these regions. See [[7.0_liftoff_hg19]])
# didn't end up going this direction actually...
module load bedtools2
bedtools getfasta -fi $REF -bed $BED > /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/CDS.fasta
```
GCA is GenBank and GCF is RefSeq



FIle locations
```
REF=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/GCF_018345385.1_ASM1834538v1_genomic.fna

GFF=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/genomic.gff

# has lengths of all scaffolds
INDEX=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/GCF_018345385.1_ASM1834538v1_genomic.fna.fai

# bed file of autosomes
BED=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.bed

# mappability bed file - genmap
MAPMASK=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/genmap/artic/GCF_018345385.1_ASM1834538v1_genomic_E2_high_mapability_mask.bed
```

>lcl|NC_054824.1_cds_XP_041619799.1_1 [gene=LOC121495839] [db_xref=GeneID:121495839] [protein=KATNB1-like protein 1] [protein_id=XP_041619799.1] [location=complement(281572..282486)] [gbkey=CDS]

ATGGCATCTGAAGCCCACAATGTTAAAAAACGAACCTTTTGTAATAATATTGAGGATCACTCCACTGATCTTCCTAGAAA