#!/bin/bash -l
#SBATCH --job-name=mpileup
#SBATCH --array=1-24
#SBATCH --time 24:00:00
#SBATCH --mem=500M
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/mpileup_variant_%a.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/mpileup_variant_%a.err

start=`date +%s`
echo "My SLURM_ARRAY_TASK_ID: " $SLURM_ARRAY_TASK_ID

# fail on errors and don't allow unset variables
set –o pipefail
set –o errexit
set –o nounset

WORKDIR=/group/ctbrowngrp2/cbquinn/fox4

BAMLIST="$WORKDIR/0_bams/bamlist_42.txt"
INDIR="$WORKDIR/0_bams/indelrealigned"
OUTDIR="$WORKDIR/1_vcfs/variant"
CHR=$(sed "${SLURM_ARRAY_TASK_ID}q;d" /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.bed | cut -f1)
REF=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/GCF_018345385.1_ASM1834538v1_genomic.fna

echo "Calling variant sites for chromosome: $CHR (chr${SLURM_ARRAY_TASK_ID})"

# the -G flag instructs bcftools to conduct single-sample calling (i.e., no assumption of HWE)
# -v indicates variant sites only

module load bcftools/1.14
#module load deprecated/bcftools/1.16 # this is is deprecated on the new farm 

bcftools mpileup -f $REF \
    --max-depth 2000 \
    -q 30 --min-BQ 30 \
    -a DP,AD,ADF,ADR,INFO/AD \
    -r ${CHR} \
    -b $BAMLIST | \
bcftools call -G - -m -v -O b -o $OUTDIR/n41/unfiltered/bcf_variant_unfiltered_42_chr${SLURM_ARRAY_TASK_ID}.bcf

end=`date +%s`
runtime=$((end-start))
echo "finished!"
echo $runtime



Rename samples
```
bcftools query --list-samples --format "[%SAMPLE\n]" $FILE
```
