
## my way

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/window_het.sh
```sh
#!/bin/bash -l
#SBATCH --job-name=het
#SBATCH --array=1-27%15
#SBATCH --time 2-00:00:00
#SBATCH --mem=4GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/window_het_%a.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/window_het_%a.err

LIST=/group/ctbrowngrp2/cbquinn/fox4/0_bams/samplelist_27_wdepths10xplus.txt
SAMPLE=$(sed "${SLURM_ARRAY_TASK_ID}q;d" $LIST | cut -f1)
INDIR=/group/ctbrowngrp3/scratch/cbquinn/fox/vcfs/invariant
BCF=$INDIR/bcf_invariant_filtered_${SAMPLE}.bcf
BED=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes_1MBwindows.bed
OUTDIR=/group/ctbrowngrp2/cbquinn/fox4/2_heterozygosity/windows/1MB/

module load bcftools/1.14
module load bedtools2/2.30.0

cd $OUTDIR

for CHR in NC_054824.1 NC_054825.1 NC_054826.1 NC_054827.1 NC_054828.1 NC_054829.1 NC_054830.1 NC_054831.1 NC_054832.1 NC_054833.1 NC_054834.1 NC_054835.1 NC_054836.1 NC_054837.1 NC_054838.1 NC_054839.1 NC_054840.1 NC_054841.1 NC_054842.1 NC_054843.1 NC_054844.1 NC_054845.1 NC_054846.1 NC_054847.1

do

sed -n "/^$CHR/p" $BED > ${SAMPLE}.${CHR}.bed

# create bed file of hets and then intersect with windows bedfile
bcftools view -r $CHR $BCF | \
bcftools query -f '%CHROM\t%POS\t[%GT]\n' | \
awk '$3 == "0/1" {print}' | \
awk -v OFS="\t" '{print $1, $2-1, $2, $3}' | \
bedtools intersect -a ${SAMPLE}.${CHR}.bed -b - -c -sorted > ${SAMPLE}.${CHR}.hets.txt

# create bed file of non-missing sites and then intersect with windows bedfile
# very large "-b" files cause bedtools to soak up huge amounts of memory. Try running this by itself with increased memory and with the "sorted" algorithm
bcftools view -r $CHR $BCF | \
bcftools query -f '%CHROM\t%POS\t[%GT]\n' | \
awk '$3 != "./." {print}' | \
awk -v OFS="\t" '{print $1, $2-1, $2, $3}' | \
bedtools intersect -a ${SAMPLE}.${CHR}.bed -b - -c -sorted > ${SAMPLE}.${CHR}.sites.txt 

paste ${SAMPLE}.${CHR}.hets.txt ${SAMPLE}.${CHR}.sites.txt | awk -v OFS="\t" '{print $1, $2, $3, $4, $8}' > ${SAMPLE}.${CHR}.combined.tsv

done

cat ${SAMPLE}.*.combined.tsv > ${SAMPLE}.genomewidehet_windows.tsv
```

Next in R...
We removed windows with fewer than 80% of sites called, as well as windows below the 5th percentile of the total number of calls
perc5 <- quantile(0.05, file$sites)
df %>%
filter(sites < perc5) %>%
filter(ifelse( sites < 800000))



## robinson way

-----------------------------



- how to filter?
- try Robinson window script?


Create environment that has pysam
```
conda create -n windowhet
conda activate windowhet
conda install -c bioconda pysam -n windowhet
```



get chr list
```
cut -f1 /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.bed | tr '\n' ' ' 
```
make chr lengths file
```
cut -f1,3 /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.bed > /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.lengths

```

create conda environment
```
conda create -n slidingwindowhet
conda activate slidingwindowhet
conda install htslib -n mpileup
conda install samtools -n mpileup
conda install bcftools -n mpileup
```




	/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/window_het.sh
```
#!/bin/bash -l
#SBATCH --job-name=het
#SBATCH --array=19
#SBATCH --time 24:00:00
#SBATCH --mem=500M
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/window_het_%a.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/window_het_%a.err


# python ~/bin/SlidingWindowHet.py [vcf] [chrom_lengths] [window size] [step size] [chromosome/scaffold name]

LIST=/group/ctbrowngrp2/cbquinn/fox4/0_bams/samplelist_27_wdepths10xplus.txt
SAMPLE=$(sed "${SLURM_ARRAY_TASK_ID}q;d" $LIST | cut -f1)
INDIR=/group/ctbrowngrp3/scratch/cbquinn/fox/vcfs/invariant
BCF=$INDIR/bcf_invariant_filtered_${SAMPLE}.bcf
VCF=$INDIR/bcf_invariant_filtered_${SAMPLE}.vcf.gz
CHR_LENGTHS=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.lengths
OUTDIR=/group/ctbrowngrp2/cbquinn/fox4/2_heterozygosity/windows/1MB/

conda activate windowhet
#module load deprecated/pysam/0.9.0
module load bcftools/1.14

# convert bcf to vcf
bcftools view $BCF -o $VCF -O z
bcftools index $VCF

cd $OUTDIR

for CHR in NC_054824.1 NC_054825.1 NC_054826.1 NC_054827.1 NC_054828.1 NC_054829.1 NC_054830.1 NC_054831.1 NC_054832.1 NC_054833.1 NC_054834.1 NC_054835.1 NC_054836.1 NC_054837.1 NC_054838.1 NC_054839.1 NC_054840.1 NC_054841.1 NC_054842.1 NC_054843.1 NC_054844.1 NC_054845.1 NC_054846.1 NC_054847.1
do

python ~/bin/SlidingWindowHet.py $VCF $CHR_LENGTHS 1000000 1000000 $CHR $SAMPLE

done
```

/group/ctbrowngrp3/scratch/cbquinn/fox/vcfs/invariant/bcf_invariant_filtered_RM_S18-2063.vcf.gz
58817155

~/bin/SlidingWindowHet.py is where copy of jarobin script is
```
# Script to count number of called genotypes and number of heterozygotes per sample in
# sliding windows.
#
# Requires pysam Python module (https://github.com/pysam-developers/pysam)
#
# Usage:
# python ./SlidingWindowHet.py [vcf] [chrom_lengths] [window size] [step size] [chromosome/scaffold name]
#
# Input file is a single- or multi-sample VCF file that has been filtered (passing sites
# must have "PASS" in the FILTER column) and compressed with gzip/bgzip.
#
# Two-column tab-delimited file with chromosome names and their lengths required.
#
# Windows will be non-overlapping if step size == window size.
#
# Example usage:
# python ./SlidingWindowHet.py input.vcf.gz chrom_lengths.txt 1000000 100000 chr1

import sys


import os
import gzip


# Open input file and make sure the VCF file is indexed (if not, create index)
filename = sys.argv[1]
VCF = gzip.open(filename, 'r')

if not os.path.exists("%s.tbi" % filename):
    pysam.tabix_index(filename, preset="vcf")
parsevcf = pysam.Tabixfile(filename)


# Set variables
chrom_lengths = sys.argv[2]
window_size = int(sys.argv[3])
step_size = int(sys.argv[4])
chrom = sys.argv[5]
samples = sys.argv[6]

# Generate a dictionary with chromosomes and chromosome lengths
cc=open(chrom_lengths, 'r')
chrom_size={line.strip().split('\t')[0]:line.strip().split('\t')[1] for line in cc}
cc.close()


# Get list of samples from VCF file header
samples=[]
for line in VCF:
    if line.startswith('##'):
        pass
    else:
        for i in line.split()[9:]: samples.append(i)
        break


# Get start and end positions of chromosome
for line in VCF:
    if line[0] != '#':
        start_pos = int(line.strip().split()[1])
        end_pos = int(chrom_size[chrom])
        break


# Create output file
output = open(filename + '_het_%swin_%sstep.txt' % (window_size, step_size), 'w')
output.write('chrom\twindow_start\tsites_total\tcalls_%s\thets_%s\n' % ('\tcalls_'.join(samples), '\thets_'.join(samples)) )


# Fetch a region, ignore sites that fail filters, tally genotype calls and heterozygotes
def snp_cal(chrom,window_start,window_end):
    print("%s:%s" % (chrom,window_start))
    rows = tuple(parsevcf.fetch(region="%s:%s-%s" % (chrom, window_start, window_end), parser=pysam.asTuple()))
    sites_total=0
    calls=[0]*len(samples)
    hets=[0]*len(samples)
    for line in rows:
        if line[6]!=".": continue
        sites_total+=1
        for i in range(0,len(samples)):
            if line[i+9][:1]=='.': continue
            calls[i]+=1
            GT=line[i+9].split(':')[0]
            if '/' in GT: sp='/'
            if '|' in GT: sp='|'
            if GT.split(sp)[0]!=GT.split(sp)[1]: hets[i]+=1
    output.write('%s\t%s\t%s\t%s\t%s\n' % (chrom,window_start,sites_total,'\t'.join(map(str,calls)),'\t'.join(map(str,hets))) )


# Initialize window start and end coordinates
window_start = start_pos
window_end = start_pos+window_size-1


# Calculate stats for window, update window start and end positions,
# repeat to end of chromosome
while window_end <= end_pos:
    if window_end < end_pos:
        snp_cal(chrom,window_start,window_end)
        window_start = window_start + step_size
        window_end = window_start + window_size - 1
    else:
        snp_cal(chrom,window_start,window_end)
        break
else:
    window_end = end_pos
    snp_cal(chrom,window_start,window_end)


# Close files and exit
VCF.close()
output.close()

exit()

```