modified from [[pseudocode for shared roh]]

Step 1: create a bedfile of ROH for every individual -- only 100KB to 1MB

```bash

module load bcftools
module load bedtools2

VCF=/group/ctbrowngrp2/cbquinn/fox4/1_vcfs/variant/n34/bcf_variant_filterA_ac1.dp3.mis20_n34_masked.vcf.gz
ROHOUT=/group/ctbrowngrp2/cbquinn/fox4/3_ROH/bcftools/bcf_variant_filterA_ac1.dp3.mis20_n34_masked.bcf.ROH_RG.txt
WORKDIR=/group/ctbrowngrp2/cbquinn/fox4/3_ROH/bcftools/IBS

cd $WORKDIR

IND=$(bcftools query -l $VCF)

for i in $IND
do
awk -v var="$i" '($2 == var) && ($6 > 100000)' $ROHOUT | cut -f3,4,5 > all/${i}_ROH_all.bed
awk -v var="$i" '($2 == var) && ($6 > 100000) && ($6 < 1000000)' $ROHOUT | cut -f3,4,5 > short/${i}_ROH_short.bed
awk -v var="$i" '($2 == var) && ($6 > 1000000)' $ROHOUT | cut -f3,4,5 > long/${i}_ROH_long.bed
done
```

Step 2: create a list of everypairwise comparison: mypairs (ind1 ind2)
bcftools query -l $VCF
```
echo $IND | tr " " "\n" > allsamples.txt
```
in R...
```R
library(tidyverse)
ind <- read_tsv("allsamples.txt", col_names=c("ind")) %>%
pull(ind) 
pairs <- combn(ind, 2, paste, collapse=' ') %>%
as_tibble() %>%
separate(value, c("ind1", "ind2"), sep=" ")
	write_tsv(pairs, "pairwise_combos.txt", col_names=FALSE)
q()
```
561 pairs


this script runs with the argument of size class: short, long, or all
takes about 3 hours for short

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/ibs_roh.sh
```sh
#!/bin/bash -l
#SBATCH --job-name=rohibs
#SBATCH --nodes 1
#SBATCH --time 1-00:00:00
#SBATCH --mem=6GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/ibs_roh.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/ibs_roh.err

# argument 1 of the script is short long or all 

class=$1

module load bcftools
module load bedtools2

VCF=/group/ctbrowngrp2/cbquinn/fox4/1_vcfs/variant/n34/bcf_variant_filterA_ac1.dp3.mis20_n34_masked.vcf.gz
ROHOUT=/group/ctbrowngrp2/cbquinn/fox4/3_ROH/bcftools/bcf_variant_filterA_ac1.dp3.mis20_n34_masked.bcf.ROH_RG.txt
WORKDIR=/group/ctbrowngrp2/cbquinn/fox4/3_ROH/bcftools/IBS

cd $WORKDIR/$class

rm sharedIBSROH_${class}.summary.txt

while read -r IND1 IND2
do

# find overlap in ROH
# and subset to at least 10KB overlap
bedtools intersect -a ${IND1}_ROH_${class}.bed -b ${IND2}_ROH_${class}.bed | \
awk '$3-$2 > 10000' > ${IND1}_${IND2}.sharedROH.bed

# write temp ind file
echo -e "$IND1\n$IND2" > pair.ind

# extract shared ROH regions from vcf
# and subset to the two relevant individuals
bcftools view -R ${IND1}_${IND2}.sharedROH.bed -S pair.ind $VCF -O z -o ${IND1}_${IND2}.sharedROH.vcf.gz

# index vcf
tabix -p vcf ${IND1}_${IND2}.sharedROH.vcf.gz

# Count number of fixed differences per shared ROH region (i.e. don't count heterozygotes or missing data)
# for GATk have to replace |s with \s
while read -r CHR START END 
do
bcftools query -r ${CHR}:${START}-${END} -f '[\t%GT]\n' ${IND1}_${IND2}.sharedROH.vcf.gz | \
sed 's/|/\//g' | \
awk '$1!=$2 && $1!="./." && $2!="./."' | grep -v "0/1" | wc -l
done < ${IND1}_${IND2}.sharedROH.bed > ${IND1}_${IND2}.fixeddiff.txt

# get number of SNPs in each region
bedtools intersect -a ${IND1}_${IND2}.sharedROH.bed -b ${IND1}_${IND2}.sharedROH.vcf.gz -c | awk '//{print $4 }' > ${IND1}_${IND2}.sites.txt

# colum headers: chr, start, end, length, fixed diff bp, fixed diff proportion of overlap
# binary classification of IBS or not-IBS
# use same haplotype threshold as Robinson et al 2021

paste ${IND1}_${IND2}.sharedROH.bed ${IND1}_${IND2}.sites.txt ${IND1}_${IND2}.fixeddiff.txt | awk '{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $3-$2, $4, $5, $5/($3-$2), 0}'| \
awk '{ if($7<5e-5) $8="same"; else $8="diff"; print $0; } ' > ${IND1}_${IND2}.sharedROHandIBScat.txt

# calculate stats

# Total shared ROH length in bp 

SHARED_ROH_LEN=$(awk '{sum+=$4}END{print sum}' ${IND1}_${IND2}.sharedROHandIBScat.txt)

# Total shared ROH and same haplotype

SHARED_ROH_IBS_LEN=$(awk '$8 == "same" {sum+=$4}END{print sum}' ${IND1}_${IND2}.sharedROHandIBScat.txt)

# Shared ROH and different haplotype (frequency of fixed differences >=5e-5 over ROH length)

SHARED_ROH_nonIBS_LEN=$(awk '$8 == "diff" {sum+=$4}END{print sum}' ${IND1}_${IND2}.sharedROHandIBScat.txt)

echo -e "$IND1\t$IND2\t$SHARED_ROH_LEN\t$SHARED_ROH_IBS_LEN\t$SHARED_ROH_nonIBS_LEN" >> sharedIBSROH.summary.txt

done < ../pairwise_combos.txt
```

