
Work flow for final analysis (using filterA vcf that has MAC=1 and minDP=3). Before this I tested sensitivity of PCA, Admxiture, and ROH analyses to these filtering criteria (see scripts below) + different LD pruning levels.

#### Run 1: keep all individuals
1. convert chromsomes from scaffold names to numeric
2. create bed/bim files for plink
3. generate list of sites in LD to remove
4. make new bed/bim files that are LD pruned (input files for Admixture)
5. run PCA, create IBS matrix

#### Run 2: Downsample mountain populations
1. Create list of individuals to keep
2. remove those from the LD-pruned dataset and create new input files for admixture
3. run PCA 




Admixture requires numeric chromosomes.  
Use vcftools to convert to plink using a chr map
First make chromosome map (2 col, tab delimited: old names, new names)

```
paste -d "\t" <(cut -f1 /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/autosomes.bed) <(seq 24) > /group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/chrmap_numeric.tsv
```
/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/plink_LDprune.sh
```
#!/bin/bash -l
#SBATCH --job-name=LDprune
#SBATCH --time 06:00:00
#SBATCH --mem=12GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/plink_LDprune.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/plink_LDprune.err

INDIR=/group/ctbrowngrp2/cbquinn/fox4/1_vcfs/variant/n34/final
REMOVE_LIST=/group/ctbrowngrp2/cbquinn/fox4/1_vcfs/variant/n34/KING/firstorder.remove
CHR_MAP=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/chrmap_numeric.tsv

module load deprecated/plink/1.90
module load vcftools/0.1.14
module load bcftools/1.14

cd $INDIR

i=bcf_variant_filterA_ac1.dp3.mis20_n34_masked.vcf.gz

VCF=../$i
OUT=${i%.vcf.gz}

# rename chromosomes to be numeric (admixture requires)
bcftools view -O z $VCF | 
vcftools --gzvcf - --chrom-map $CHR_MAP --plink --out ${OUT}

# create plink bed and bim files 
# (set-missing-var-ids is necessary or it produces a snp list filled with ".")
plink --file ${OUT} \
    --allow-extra-chr \
    --chr-set 24 \
    --make-bed \
    --set-missing-var-ids @:# \
    --out ${OUT}

# run IBS on full non-pruned dataset
plink --bfile ${OUT} \
    --allow-extra-chr \
    --chr-set 24 \
    --distance square ibs \
    --out ${OUT}_LDpruned

# create list of LD snps
plink --bfile ${OUT} \
	--allow-extra-chr \
	--chr-set 24 \
	--indep-pairwise 100 10 0.1 --out ${OUT}_LDpruned

# create new plink files from LD-pruned dataset
# (keeps 398,245 sites)
plink --bfile ${OUT} \
    --allow-extra-chr \
    --chr-set 24 \
    --make-bed \
    --extract ${OUT}_LDpruned.prune.in \
    --recode \
    --out ${OUT}_LDpruned

# run pca and IBS on pruned set
# --cluster and --matrix print an IBS matrix
plink --bfile ${OUT}_LDpruned \
    --allow-extra-chr \
    --chr-set 24 \
    --pca 34 --cluster \
    --matrix \
    --out ${OUT}_LDpruned

# try running IBS a different way
plink --bfile ${OUT}_LDpruned \
    --allow-extra-chr \
    --chr-set 24 \
    --distance square ibs allele-ct \
    --out ${OUT}_LDpruned

# (they work the same way)

# rerun on reduced sample set 
plink --bfile ${OUT}_LDpruned \
    --allow-extra-chr \
    --chr-set 24 \
    --keep ../reduced_sampleset_structure.txt \
    --pca 16 --cluster \
    --make-bed \
    --out ${OUT}_LDpruned_reduced

# ibs on reduced sample set
plink --bfile ${OUT}_LDpruned \
    --allow-extra-chr \
    --chr-set 24 \
    --keep ../reduced_sampleset_structure.txt \
    --distance square ibs allele-ct \
    --out ${OUT}_LDpruned_reduced




scontrol show job ${SLURM_JOB_ID}
sstat --format 'JobID,MaxRSS,AveCPU' -P ${SLURM_JOB_ID}.batch
```


