
## 1. download outgroups

species | ncbi | reference
--- | ---  | ---
Racoon dog (Nyctereutes procyonoides) | GCA_905146905.1 | Lan et al 2022
Sand fox (Vulpes ferrilata) | GCA_024500485.1 | Lyu et al 2022

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/curl_raccoondog_ref.sh
```
#!/bin/bash -l
#SBATCH --job-name=racc
#SBATCH --time 01:00:00
#SBATCH --mem=500M
#SBATCH -p bmh
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/curl_raccoondog_ref.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/curl_raccoondog_ref.err


cd /group/ctbrowngrp3/scratch/cbquinn/fox/refs
curl -OJX GET "https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCA_905146905.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA,SEQUENCE_REPORT&filename=GCA_905146905.1.zip" -H "Accept: application/zip"

unzip GCA_905146905.1.zip
mv /group/ctbrowngrp3/scratch/cbquinn/fox/refs/ncbi_dataset/data/GCA_905146905.1 /group/ctbrowngrp3/scratch/cbquinn/fox/refs/
rm -r /group/ctbrowngrp3/scratch/cbquinn/fox/refs/ncbi_dataset/

# create .fai index file
cd GCA_905146905.1/
samtools faidx GCA_905146905.1_NYPRO_anot_genome_genomic.fna

module load bwa
bwa index GCA_905146905.1_NYPRO_anot_genome_genomic.fna
```


/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/curl_sandfox_ref.sh
```
#!/bin/bash -l
#SBATCH --job-name=sand
#SBATCH --time 01:00:00
#SBATCH --mem=500M
#SBATCH -p bmh
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/curl_sandfox_ref.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/curl_sandfox_ref.err

cd /group/ctbrowngrp3/scratch/cbquinn/fox/refs
curl -OJX GET "https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCA_024500485.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA,SEQUENCE_REPORT&filename=GCA_024500485.1.zip" -H "Accept: application/zip"

unzip GCA_024500485.1.zip
mv /group/ctbrowngrp3/scratch/cbquinn/fox/refs/ncbi_dataset/data/GCA_024500485.1 /group/ctbrowngrp3/scratch/cbquinn/fox/refs/
rm -r /group/ctbrowngrp3/scratch/cbquinn/fox/refs/ncbi_dataset/

# create .fai index file
cd GCA_024500485.1/
samtools faidx GCA_024500485.1_ASM2450048v1_genomic.fna

module load bwa
bwa index GCA_024500485.1_ASM2450048v1_genomic.fna
```

## chop references into tiny pieces and align 
following methods of Kardos et al 2023: Inbreeding depression explains killer whale population dynamics

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/align_outgroup_refs.sh $1
```
#!/bin/bash -l
#SBATCH --job-name=chop
#SBATCH --time 3-00:00:00
#SBATCH --mem=8GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/align_outgroup_refs.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/align_outgroup_refs.err

# make things fail on errors
set -o nounset
set -o errexit

module load bedtools/2.27.1
module load seqtk/1.3r106
module load bwa
module load samtools/1.16

cd /group/ctbrowngrp3/scratch/cbquinn/fox/refs/

SPP=$1
ARCTIC=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/GCF_018345385.1_ASM1834538v1_genomic.fna

if [ $SPP == "dog" ]
then
REF=/group/ctbrowngrp2/cbquinn/wolf2/tempref/canFam3_withY.fa
OUTDIR=dog/
fi
if [ $SPP == "raccoondog" ]
then
REF=/group/ctbrowngrp3/scratch/cbquinn/fox/refs/GCA_905146905.1/GCA_905146905.1_NYPRO_anot_genome_genomic.fna
OUTDIR=GCA_905146905.1/
fi
if [ $SPP == "sandfox" ]
then
REF=/group/ctbrowngrp3/scratch/cbquinn/fox/refs/GCA_024500485.1/GCA_024500485.1_ASM2450048v1_genomic.fna
OUTDIR=GCA_024500485.1/
fi

echo "cutting up Ref: ${REF}"

bedtools makewindows -g ${REF}.fai -w 70 -s 10 | \
bedtools getfasta -fi $REF -bed - -fo ${OUTDIR}${SPP}.70_10_windows.fa

seqtk seq -F '#' ${OUTDIR}${SPP}.70_10_windows.fa | gzip > ${OUTDIR}${SPP}.70_10_windows.fq.gz

bwa mem ${ARCTIC} ${OUTDIR}${SPP}.70_10_windows.fq.gz | \
samtools view -h -b - | \
samtools sort -o ${OUTDIR}${SPP}.70_10_mapped2arctic.bam -

echo "adding readgroups..."
module load picardtools/2.7.1

java -jar ${PICARD}/picard.jar AddOrReplaceReadGroups \
    I=${OUTDIR}${SPP}.70_10_mapped2arctic.bam \
    O=${OUTDIR}${SPP}.70_10_mapped2arctic_readgroups.bam  \
    RGID=${SPP} RGPL=ref RGPU=NULL RGSM=${SPP} RGLB=lib \
    VALIDATION_STRINGENCY=LENIENT

# remove raw output bam
rm ${OUTDIR}${SPP}.70_10.bam
```

## call outgroup alleles at variant sites 
Use bed file to call sites for reference genomes only here

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/call_outgroup_refs.sh
```shell
#!/bin/bash -l
#SBATCH --job-name=mpileup
#SBATCH --time 2-00:00:00
#SBATCH --mem=8GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/call_outgroup_refs.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/call_outgroup_refs.err


module load bedtools2
module load bcftools/1.14
module load samtools
module load tabix

VCF=/group/ctbrowngrp2/cbquinn/fox4/1_vcfs/variant/n34/bcf_variant_filterA_ac1.dp3.mis20_n34_masked.vcf.gz

INDIR=/group/ctbrowngrp3/scratch/cbquinn/fox/refs
OUTDIR=/group/ctbrowngrp2/cbquinn/fox4/5_load/polarize

REF=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/GCF_018345385.1_ASM1834538v1_genomic.fna

# get variant sites from vcf
bcftools view -H $VCF | \
awk -v OFS=$'\t' '{print $1, $2}' | \
bgzip > $OUTDIR/filterA_all.sites.gz
#tabix $OUTDIR/filterA_all.sites.gz


# use mpileup to call
find "$INDIR" -type f -name "*.bam" | tr " " "\t" > $OUTDIR/outgroup.bamlist

#for bam in $(find "$INDIR" -type f -name "*.bam")
#do
#samtools index $bam
#done

bcftools mpileup -f $REF \
    -q 0 --min-BQ 0 \
    -a DP,AD,ADF,ADR,INFO/AD \
    -T $OUTDIR/filterA_all.sites.gz \
    -b $OUTDIR/outgroup.bamlist | \
bcftools call -m -O b -o $OUTDIR/filterA_outgroups.bcf
```

## filter variant alleles
/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/filter_outgroup_refs.sh
```shell
#!/bin/bash -l
#SBATCH --job-name=mpileup
#SBATCH --time 05:00:00
#SBATCH --mem=4GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/filter_outgroup_refs.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/filter_outgroup_refs.err


module load bcftools
cd /group/ctbrowngrp2/cbquinn/fox4/5_load/polarize

# get rid of triallelic snps and indels
bcftools view -M 2 filterA_outgroups.bcf | \
# filter: turn hets to ./. (mismapping)
bcftools +setGT -- -t q -i 'GT="het"' -n "./."  | \
# filter: get rid of sites with 2 or 3 samples missing and get rid of indels
bcftools view -e 'N_MISSING > 1' | \
bcftools query -f '%CHROM %POS %INDEL %REF %ALT [ %GT]\n' | \
# recode 1/1 as alt allele and 0/0 as ref allele
sed 's|0/0|0|g' | sed 's|1/1|1|g' | sed 's|./.|NA|g' | \
# remove indels
awk '{if ($3 != 1) {print $0}}' > filterA_outgroups_temp.txt

# remove sites that are within 3 bp of each other
awk '
    NR == 1{old = $2; next}     # if 1st line 
    {print $2 - old; old = $2}  # else...
' filterA_outgroups_temp.txt > temp.dist

paste filterA_outgroups_temp.txt <(sed '1s/^/0\n/' temp.dist) temp.dist | \
tr -s " " "\t" | \
awk '{ if( ($9 >= 3) && ($10 >= 3) ) { print } }' | \
bgzip > filterA_outgroups_filt.txt.gz


#rm temp.dist
#rm *_temp.txt
```




determine ancestral allele in R
```
unset CONDA_EXE
module load R/3.6.3
R
```

```r
library(tidyverse)
d <- read.table("filterA_outgroups_filt.txt.gz") %>%
as_tibble()
names(d) <- c("chr", "pos", "indel", "ref", "alt", "dog", "raccoondog", "sandfox", "dist1", "dist2")

# recode...
# number of sites with data (2 or 3)
# sum across alleles...
# if sum = 0 ~ ancestral = ref
# if sum = 1 & total = 3 ~ ancestral = ref
# if sum=2 & total =3 ~ ancestal = alt
# if sum = 3, ancestral = alt

0 0 0 0 ref
0 0 0 ref
0 0 0 1 ref
0 0 1 ref
0 1 1 alt
0 0 1 1 NA
0 1 1 1 alt


count_na_func <- function(x) sum(is.na(x)) 

df <- d %>%
select(-dist1, -dist2, -indel) %>%
mutate(sum = rowSums(across(c(dog, raccoondog, sandfox)), na.rm=TRUE),
	  total = 3-apply(., 1, count_na_func)
	  ) %>%
		 mutate(
		 ref=as.character(ref),
		 alt=as.character(alt)) %>%
	mutate(AA=case_when(
	sum == 0 ~ ref,
	sum == 1 ~ ref,
	sum == 2 & total == 2 ~ alt,
	sum == 2 & total == 3 ~ as.character("NA"),
	sum == 3 ~ alt
	)) %>%
	filter(AA != "NA")%>%
	mutate(polarity=ifelse(AA==ref, "ref", "reverse"))

df.summary <- df %>%
group_by(polarity) %>%
count()

write_tsv(df, "ancestralalleles.df", col_names=FALSE)
write_tsv(df.summary, "polarity.switches.txt", col_names=FALSE)

# convert to bed file
bed <- df %>%
	mutate(start=pos-1) %>%
	rename(end=pos) %>%
	select(chr, start, end, AA, polarity)

write_tsv(bed, "ancestralalleles.bed", col_names=FALSE)
	  
```



```
# to clear r workspace
ls 
rm(list = ls())

```
gzip and index large files
```
bgzip ancestralalleles.df > ancestralalleles.df.gz
bgzip ancestralalleles.bed > ancestralalleles.bed.gz
tabix ancestralalleles.bed.gz
```
