

function to sample alleles

/group/ctbrowngrp2/cbquinn/fox4/rscripts/sfsfunctions.R
```r
# two functions that work together to get allele counts for each population
# but because sample sizes are unequal, function randomly samples chromosomes without replacement from each population
# randomly sample n*2 chr for each pop

sample_alleles <- function(mysite, df, sample.size){
  df %>%
    filter(site == mysite, !is.na(geno)) %>%
    separate(geno, into=c("chrA", "chrB"), sep="/") %>%
    select(CHROM, POS, site, chrA, chrB, sampleID, pop2) %>%
    pivot_longer(c("chrA", "chrB"), names_to="chr", values_to="allele") %>%
    mutate(allele=as.numeric(allele)) %>%
    group_by(pop2) %>%
    mutate(n=n()) %>%
    filter(n >= sample.size*2) %>%
    sample_n(sample.size*2, replace=FALSE) %>%
    group_by(pop2) %>%
    summarize(allele.count=sum(allele))  %>%
    mutate(site=mysite)
}


get_ACsite_per_pop <- function(vcf_file, exclude="none", sample.size){
  
  vcf <- read.vcfR(vcf_file)
  
  if(exclude[1] != "none"){
    myind <- colnames(vcf@gt)[-1]  
    vcf@gt<- vcf@gt[,c(TRUE, !(myind %in% exclude))]
    
    # omit monomorphic sites 
    vcf <- vcf[is.polymorphic(vcf, na.omit=TRUE)]
  }
  
  # convert to tidy format
  tidy_vcf <- vcfR2tidy(vcf,
                        format_fields=c("GT"),
                        dot_is_NA=TRUE)
  
  site.df <- tidy_vcf$fix %>%
    unite(site, c("CHROM", "POS"), sep = "-", remove = FALSE) 
  
  geno.df <- tidy_vcf$gt %>% 
    inner_join(site.df) %>%
    rename(sampleID=Indiv, geno=gt_GT) %>%
    left_join(info, by="sampleID")
  
  # use sample allele function to randomly sample alleles for every site that meets criteria
  ac.list <- lapply(site.df$site, function(x){
    sample_alleles(mysite=x, df=geno.df, sample.size=sample.size)
  })
  
  ac.df <- bind_rows(ac.list) 
  
  return(ac.df)
}

# identical to above except they don't group by population

sample_alleles_total <- function(mysite, df, sample.size){
  temp <- df %>%
    filter(site == mysite, !is.na(geno)) %>%
    separate(geno, into=c("chrA", "chrB"), sep="/") %>%
    select(CHROM, POS, site, chrA, chrB, sampleID, pop2) %>%
    pivot_longer(c("chrA", "chrB"), names_to="chr", values_to="allele") %>%
    mutate(allele=as.numeric(allele)) %>%
    mutate(n=n()) %>%
    filter(n >= sample.size*2) %>%
    sample_n(sample.size*2, replace=FALSE) 
    
    bind_cols(
    temp %>%
    summarize(allele.count=sum(allele))  ,
	temp %>%
	summarize(allele.total=n()) %>%
    mutate(site=mysite)
	)
}


get_ACsite_total <- function(vcf_file, exclude="none", sample.size){
  
  vcf <- read.vcfR(vcf_file)
  
  if(exclude[1] != "none"){
    myind <- colnames(vcf@gt)[-1]  
    vcf@gt<- vcf@gt[,c(TRUE, !(myind %in% exclude))]
    
    # omit monomorphic sites 
    vcf <- vcf[is.polymorphic(vcf, na.omit=TRUE)]
  }
  
  # convert to tidy format
  tidy_vcf <- vcfR2tidy(vcf,
                        format_fields=c("GT"),
                        dot_is_NA=TRUE)
  
  site.df <- tidy_vcf$fix %>%
    unite(site, c("CHROM", "POS"), sep = "-", remove = FALSE) 
  
  geno.df <- tidy_vcf$gt %>% 
    inner_join(site.df) %>%
    rename(sampleID=Indiv, geno=gt_GT) %>%
    left_join(info, by="sampleID")
  
  # use sample allele function to randomly sample alleles for every site that meets criteria
  ac.list <- lapply(site.df$site, function(x){
    sample_alleles_total(mysite=x, df=geno.df, sample.size=sample.size)
  })
  
  ac.df <- bind_rows(ac.list) 
  
  return(ac.df)
}
```

/group/ctbrowngrp2/cbquinn/fox4/rscripts/create_sfs_random.R
```R
library(tidyverse)
library(vcfR)

args = commandArgs(trailingOnly = TRUE)
sample.size <- as.numeric(args[1])

source("/group/ctbrowngrp2/cbquinn/fox4/rscripts/sfsfunctions.R")

# read in sample meta data
info <- read_tsv("/group/ctbrowngrp2/cbquinn/fox4/0_bams/info.tsv") %>%
  select(SeqName1, SeqName2, lineage, pop2, in_ms, depth_lagopus_mapq30) %>%
  rename(sampleID=SeqName2, depth=depth_lagopus_mapq30, pop=pop2) %>%
  filter(in_ms == "yes") %>%
  select(-in_ms) %>%
  mutate(pop2 = case_when(
    lineage == "east" ~ "EAST",
    lineage != "east" ~ pop )) %>%
  mutate(lineage=factor(lineage, levels=c("eurasia", "north", "east", "west")),
         pop2=factor(pop2, levels = c("RUS", "AK", "EAST", "SN", "LAS", "ORC", "WAC", "RM", "SV"))) %>%
  arrange(lineage, pop, sampleID) %>%
  mutate(sampleID=factor(sampleID, levels=unique(sampleID)))

myvcfs <- c(
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/lof.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/sift/annotation_uniprot/missense_del.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/synonymous.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/intergenic.vcf.gz"
)

ac.list <- lapply(myvcfs, function(x){
  get_ACsite_per_pop(vcf_file=x, exclude=c("AK_S12-1159", "RUS_S12-0237"), sample.size=sample.size)
})

names(ac.list) <- basename(myvcfs)
sfs <- ac.list %>% 
  bind_rows(., .id="myfile") %>%
  mutate(group=gsub(".vcf.gz", "", myfile)) %>%
  mutate(group = factor(group, levels=c("synonymous", "missense_del", "lof"))) %>%
  select(-myfile)

# remove sites where not all populations have sufficient sample size
sfs_nomiss <- sfs %>%
  pivot_wider(names_from="pop2", values_from="allele.count") %>%
  na.omit() %>%
  pivot_longer(-c("group", "site"), values_to="allele.count", names_to="pop2")

write_tsv(sfs, paste0("sfs_n", sample.size,"wsites.tsv"))
write_tsv(sfs_nomiss, paste0("sfs_n", sample.size,"wsites_nomissing.tsv"))
```

/group/ctbrowngrp2/cbquinn/fox4/rscripts/create_sfs_fixed.sh
```R
# same as other script but restricts RM chr sampling to those in Oregon region

library(tidyverse)
library(vcfR)

args = commandArgs(trailingOnly = TRUE)
sample.size <- as.numeric(args[1])
print("sample size is:")
print(sample.size)


source("/group/ctbrowngrp2/cbquinn/fox4/rscripts/sfsfunctions.R")

# read in sample meta data
info <- read_tsv("/group/ctbrowngrp2/cbquinn/fox4/0_bams/info.tsv") %>%
  select(SeqName1, SeqName2, lineage, pop2, in_ms, depth_lagopus_mapq30) %>%
  rename(sampleID=SeqName2, depth=depth_lagopus_mapq30, pop=pop2) %>%
  filter(in_ms == "yes") %>%
  select(-in_ms) %>%
  mutate(pop2 = case_when(
    lineage == "east" ~ "EAST",
    lineage != "east" ~ pop )) %>%
  mutate(lineage=factor(lineage, levels=c("eurasia", "north", "east", "west")),
         pop2=factor(pop2, levels = c("RUS", "AK", "EAST", "SN", "LAS", "ORC", "WAC", "RM", "SV"))) %>%
  arrange(lineage, pop, sampleID) %>%
  mutate(sampleID=factor(sampleID, levels=unique(sampleID)))

myvcfs <- c(
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/lof.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/sift/annotation_uniprot/missense_del.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/synonymous.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/intergenic.vcf.gz"
)

ac.list <- lapply(myvcfs, function(x){
  get_ACsite_per_pop(vcf_file=x, exclude=c("AK_S12-1159", "RUS_S12-0237", "RM_S13-3312", "RM_S13-3301"
), sample.size=sample.size)
})

names(ac.list) <- basename(myvcfs)
sfs <- ac.list %>% 
  bind_rows(., .id="myfile") %>%
  mutate(group=gsub(".vcf.gz", "", myfile)) %>%
  mutate(group = factor(group, levels=c("synonymous", "missense_del", "lof"))) %>%
  select(-myfile)

# remove sites where not all populations have sufficient sample size
sfs_nomiss <- sfs %>%
  pivot_wider(names_from="pop2", values_from="allele.count") %>%
  na.omit() %>%
  pivot_longer(-c("group", "site"), values_to="allele.count", names_to="pop2")

write_tsv(sfs, paste0("sfs_n", sample.size,"_fixed_wsites.tsv"))
write_tsv(sfs_nomiss, paste0("sfs_n", sample.size,"_fixed_wsites_nomissing.tsv"))
```

/group/ctbrowngrp2/cbquinn/fox4/rscripts/create_sfs_total.R
```r
library(tidyverse)
library(vcfR)

args = commandArgs(trailingOnly = TRUE)
sample.size <- as.numeric(args[1])
print("sample size is:")
print(sample.size)

source("/group/ctbrowngrp2/cbquinn/fox4/rscripts/sfsfunctions.R")

# read in sample meta data
info <- read_tsv("/group/ctbrowngrp2/cbquinn/fox4/0_bams/info.tsv") %>%
  select(SeqName1, SeqName2, lineage, pop2, in_ms, depth_lagopus_mapq30) %>%
  rename(sampleID=SeqName2, depth=depth_lagopus_mapq30, pop=pop2) %>%
  filter(in_ms == "yes") %>%
  select(-in_ms) %>%
  mutate(pop2 = case_when(
    lineage == "east" ~ "EAST",
    lineage != "east" ~ pop )) %>%
  mutate(lineage=factor(lineage, levels=c("eurasia", "north", "east", "west")),
         pop2=factor(pop2, levels = c("RUS", "AK", "EAST", "SN", "LAS", "ORC", "WAC", "RM", "SV"))) %>%
  arrange(lineage, pop, sampleID) %>%
  mutate(sampleID=factor(sampleID, levels=unique(sampleID)))

myvcfs <- c(
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/lof.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/sift/annotation_uniprot/missense_del.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/synonymous.vcf.gz",
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/intergenic.vcf.gz"
)

ac.list <- lapply(myvcfs, function(x){
  get_ACsite_total(vcf_file=x, exclude=c("AK_S12-1159", "RUS_S12-0237"), sample.size=sample.size)
})

names(ac.list) <- basename(myvcfs)
sfs <- ac.list %>% 
  bind_rows(., .id="myfile") %>%
  mutate(group=gsub(".vcf.gz", "", myfile)) %>%
  mutate(group = factor(group, levels=c("intergenic", "synonymous", "missense_del", "lof"))) %>%
  select(-myfile)

# remove sites where not all populations have sufficient sample size
sfs_nomiss <- sfs %>%
  pivot_wider(names_from="pop2", values_from="allele.count") %>%
  na.omit() %>%
  pivot_longer(-c("group", "site"), values_to="allele.count", names_to="pop2")

write_tsv(sfs, paste0("sfs_n", sample.size,"_total_wsites.tsv"))
write_tsv(sfs_nomiss, paste0("sfs_n", sample.size,"_total_wsites.tsv"))

```


/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/sample_random_sfs.sh
```sh
#!/bin/bash -l
#SBATCH --job-name=sfs
#SBATCH --time 12:00:00
#SBATCH --mem=4GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/sfs.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/sfs.err

unset CONDA_EXE
module load R/3.6.3
N=${1}

echo "running with sample.size = $1"
srun Rscript /group/ctbrowngrp2/cbquinn/fox4/rscripts/create_sfs_random.R $N &> sfs.${N}.random.log
```

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/sample_fixedRM_sfs.sh
```sh
#!/bin/bash -l
#SBATCH --job-name=sfs
#SBATCH --time 12:00:00
#SBATCH --mem=4GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/sfs.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/sfs.err

unset CONDA_EXE
module load R/3.6.3

srun Rscript /group/ctbrowngrp2/cbquinn/fox4/rscripts/create_sfs_fixed.sh $1 &> sfs.${1}.fixed.log
```

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/sample_total_sfs.sh
```
#!/bin/bash -l
#SBATCH --job-name=sfs
#SBATCH --time 12:00:00
#SBATCH --mem=4GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/sfs_total.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/sfs_total.err

unset CONDA_EXE
module load R/3.6.3
N=${1}

echo "running with sample.size = $1"
srun Rscript /group/ctbrowngrp2/cbquinn/fox4/rscripts/create_sfs_total.R $N &> sfs.${N}.total.log

```


# for testing
myvcfs <- c(
"/group/ctbrowngrp2/cbquinn/fox4/5_load/snpeff/lof.vcf.gz")
sample.size <- 5

/group/ctbrowngrp2/cbquinn/fox4/rscripts/create_sfs_n5random.sh


