
## Create plink input files

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/make_gone_inputs.sh
```sh
#!/bin/bash -l
#SBATCH --job-name=gone_in
#SBATCH --time 24:00:00
#SBATCH --mem=6GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/make_gone_inputs.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/make_gone_inputs.err

OUTDIR=/group/ctbrowngrp2/cbquinn/fox4/7_GONE/inputs

VCF=/group/ctbrowngrp2/cbquinn/fox4/1_vcfs/variant/n34/bcf_variant_filterA_ac1.dp3.mis20_n34_masked.vcf.gz
CHR_MAP=/group/ctbrowngrp2/cbquinn/fox4/ref/GCF_018345385.1/chrmap_numeric.tsv


module load deprecated/plink/1.90
module load bcftools
module load vcftools/0.1.14

~/bin/Linux/

for pop in LAS ORC WAC RM
do

# filter by pop and remove monomorphic sites
# rename chromosomes to be numeric
bcftools query -l $VCF | grep "$pop" > ${pop}.names.txt
bcftools view -S ${pop}.names.txt $VCF | \
bcftools view -c 1:minor -Oz | \
vcftools --gzvcf - --chrom-map $CHR_MAP --plink --out ${pop}_filterA

# create plink files 
# (set-missing-var-ids is necessary or it produces a snp list filled with ".")
plink --file ${pop}_filterA \
    --allow-extra-chr \
    --chr-set 24 \
    --set-missing-var-ids @:# \
    --keep-allele-order \
    --recode \
    --out ${pop}_filterA
done

# do east separately - made names file by hand
pop=EAST
bcftools view -S ${pop}.names.txt $VCF | \
bcftools view -c 1:minor -Oz | \
vcftools --gzvcf - --chrom-map $CHR_MAP --plink --out ${pop}_filterA

# create plink files
plink --file ${pop}_filterA \
    --allow-extra-chr \
    --chr-set 24 \
    --set-missing-var-ids @:# \
    --keep-allele-order \
    --recode \
    --out ${pop}_filterA

# make a few subsets
# ORC without admixed individuals: ORC_7

ORC_S17-2543
ORC_S17-2544
ORC_S18-1173
ORC_S18-2062
ORC_S18-2068
ORC_S18-2069
ORC_S18-2071

plink --file ORC_filterA \
    --keep ORC_7.names.txt \
    --maf 0.01 \
    --recode \
    --out ORC7_filterA

# RM without NV
RM_S13-0649
RM_S13-2269
RM_S18-2063
RM_S13-1169
RM_S13-2260

plink --file RM_filterA \
    --keep RM5.names.txt \
    --maf 0.01 \
    --recode \
    --out RM5_filterA

# EAST with VT
ECAN_S14-0358
ECAN_S14-0407
ECAN_S14-0430

```


## Parameter file

~/bin/INPUT_PARAMETERS_FILE
kept defaults except for 
```
cMMb=0.6
Threads=12
```

## Run independent Gone replicates for each pop

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/run_gone_east.sh
```
#!/bin/bash -l
#SBATCH --job-name=east
#SBATCH --time 6-00:00:00
#SBATCH --cpus-per-task 12
#SBATCH --mem=8GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_east.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_east.err


# run replicates of the east
# ECAN individuals only (n=3), removed VT to avoid admixture-LD

module load deprecated/plink/1.90

cd /group/ctbrowngrp3/scratch/cbquinn/fox/GONE_cp2

i=1

echo "$i"

cp EAST_filterA.ped EAST_filterA_i${i}.ped
cp EAST_filterA.map EAST_filterA_i${i}.map

# submit and get jobID
JOBID=$(srun script_GONE.sh EAST_filterA_i${i} | awk '{print $4}')

while [ $i -le 20 ]
do
        
        i=$(($i+1))
		#((i++))
	        echo "$i"

cp EAST_filterA.ped EAST_filterA_i${i}.ped
cp EAST_filterA.map EAST_filterA_i${i}.map

# make script dependent on the previous
JOBID=$(srun -d afterok:$JOBID script_GONE.sh EAST_filterA_i${i} | awk '{print $4}')

done

rm EAST_filterA_i*ped
rm EAST_filterA_i*map
```


/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/run_gone_RM.sh
```
#!/bin/bash -l
#SBATCH --job-name=east
#SBATCH --time 6-00:00:00
#SBATCH --cpus-per-task 12
#SBATCH --mem=8GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_east.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_east.err


# run replicates of East

module load deprecated/plink/1.90

cd /group/ctbrowngrp3/scratch/cbquinn/fox/GONE_cp2

i=1

echo "$i"

cp EAST_filterA.ped EAST_filterA_i${i}.ped
cp EAST_filterA.map EAST_filterA_i${i}.map

# submit and get jobID
JOBID=$(srun script_GONE.sh EAST_filterA_i${i} | awk '{print $4}')

while [ $i -le 40 ]
do
        i=$(($i+1))
        echo "$i"

cp EAST_filterA.ped EAST_filterA_i${i}.ped
cp EAST_filterA.map EAST_filterA_i${i}.map

# make script dependent on the previous
JOBID=$(srun -d afterok:$JOBID script_GONE.sh EAST_filterA_i${i} | awk '{print $4}')

done

rm EAST_filterA_i*ped
rm EAST_filterA_i*map
```

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/run_gone_RM6.sh
```
#!/bin/bash -l
#SBATCH --job-name=RM
#SBATCH --time 6-00:00:00
#SBATCH --cpus-per-task 12
#SBATCH --mem=8GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_RM6.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_RM6.err


# run replicates of the RM (all ind except NV 3312)

module load deprecated/plink/1.90

cd /group/ctbrowngrp3/scratch/cbquinn/fox/GONE_cp2

plink --file RM_filterA \
    --keep /group/ctbrowngrp2/cbquinn/fox4/7_GONE/inputs/RM6.names.txt \
    --maf 0.01 \
    --recode \
    --out RM6_filterA
    
i=1
echo "$i"

# submit and get jobID
JOBID=$(srun script_GONE.sh RM6_filterA | awk '{print $4}')

while [ $i -le 20 ]
do

        echo "$i"

cp RM6_filterA.ped RM6_filterA_i${i}.ped
cp RM6_filterA.map RM6_filterA_i${i}.map

# make script dependent on the previous
JOBID=$(srun -d afterok:$JOBID script_GONE.sh RM6_filterA_i${i} | awk '{print $4}')

i=$(($i+1))
done

rm RM6_filterA_i*ped
rm RM6_filterA_i*map
```

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/run_gone_LAS.sh
```
#!/bin/bash -l
#SBATCH --job-name=LAS
#SBATCH --time 6-00:00:00
#SBATCH --cpus-per-task 12
#SBATCH --mem=8GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_LAS.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_LAS.err


# run replicates of the LAS

module load deprecated/plink/1.90

cd /group/ctbrowngrp3/scratch/cbquinn/fox/GONE_cp1
    
i=1
echo "$i"

# submit and get jobID
JOBID=$(srun script_GONE.sh LAS_filterA | awk '{print $4}')

while [ $i -le 42 ]
do

        echo "$i"

cp LAS_filterA.ped LAS_filterA_i${i}.ped
cp LAS_filterA.map LAS_filterA_i${i}.map

# make script dependent on the previous
JOBID=$(srun -d afterok:$JOBID script_GONE.sh LAS_filterA_i${i} | awk '{print $4}')

i=$(($i+1))
done

rm LAS_filterA_i*ped
rm LAS_filterA_i*map

```

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/run_gone_WAC.sh
```
#!/bin/bash -l
#SBATCH --job-name=LAS
#SBATCH --time 6-00:00:00
#SBATCH --cpus-per-task 12
#SBATCH --mem=8GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_WAC.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_WAC.err


# run replicates of the WAC

module load deprecated/plink/1.90

cd /group/ctbrowngrp3/scratch/cbquinn/fox/GONE_cp1
    
i=1
echo "$i"

# submit and get jobID
JOBID=$(srun script_GONE.sh WAC_filterA | awk '{print $4}')

while [ $i -le 42 ]
do

        echo "$i"

cp WAC_filterA.ped WAC_filterA_i${i}.ped
cp WAC_filterA.map WAC_filterA_i${i}.map

# make script dependent on the previous
JOBID=$(srun -d afterok:$JOBID script_GONE.sh WAC_filterA_i${i} | awk '{print $4}')

i=$(($i+1))
done

rm WAC_filterA_i*ped
rm WAC_filterA_i*map
```

/group/ctbrowngrp2/cbquinn/fox4/slurmscripts/run_gone_ORC7.sh
```sh
#!/bin/bash -l
#SBATCH --job-name=ORC7
#SBATCH --time 6-00:00:00
#SBATCH --cpus-per-task 12
#SBATCH --mem=8GB
#SBATCH -p bmm
#SBATCH -A ctbrowngrp
#SBATCH -o /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_ORC7.out
#SBATCH -e /group/ctbrowngrp2/cbquinn/fox4/slurmlogs/run_gone_ORC7.err


# run replicates of the ORC7 (excluding admixed 2)

module load deprecated/plink/1.90

cd /group/ctbrowngrp3/scratch/cbquinn/fox/GONE_cp2

for i in {1..20}; do 
echo $i 

cp ORC7_filterA.ped ORC7_filterA_i${i}.ped
cp ORC7_filterA.map ORC7_filterA_i${i}.map

# make script dependent on the previous
srun script_GONE.sh ORC7_filterA_i${i}

rm ORC7_filterA_i${i}.ped
rm ORC7_filterA_i${i}.map
done

```

## Plot traces in R

```R

library(tidyverse)
library(scales)
library(cowplot)

mycoldf <- data.frame(mycol=c(   "#2f2a26",
                                 "#c3c3c1",
                                 "#5c5c5c",
                                 "#ffd380",
                                 "#ffa600",
                                 "#bc5090",
                                 "#2c4875",
                                 "#8a508f",
                                 "#ff6361"
),
pop2=c("RUS", "AK", "EAST",  "SV", "RM", "WAC",  "ORC",  "LAS","SN"))


setwd("V:/3730Data/377STRs/Wildlife/Cate/WesternRedFoxWGS/round4/GONE")

myfiles.RM <- list.files(pattern="Output_Ne_RM6")
myfiles.LAS <- list.files(pattern="Output_Ne_LAS")
myfiles.WAC <- list.files(pattern="Output_Ne_WAC")
myfiles.ORC <- list.files(pattern="Output_Ne_ORC7")
myfiles <- c(myfiles.RM,myfiles.LAS, myfiles.WAC, myfiles.ORC)
mylist <- lapply(myfiles, read_tsv, skip=1)
names(mylist) <- myfiles

d <- bind_rows(mylist, .id="myfile") %>%
  separate(myfile, into=c("A", "B", "pop2", "D", "replicate") ,sep="_") %>%
  select(-A, -B, -D) 

# add in new
setwd("C:/Users/Cate/Dropbox/WGSfoxes/writing/revision1_MBE/gone")
myfiles.ECAN <- list.files(pattern="Output_Ne_ECAN")
mylist.ECAN <- lapply(myfiles.ECAN, read_tsv, skip=1)
names(mylist.ECAN) <- myfiles.ECAN
d.ECAN <- bind_rows(mylist.ECAN, .id="myfile") %>%
  separate(myfile, into=c("A", "B", "pop2", "D", "replicate") ,sep="_") %>%
  select(-A, -B, -D) 

d <- bind_rows(d, d.ECAN)

d$replicate[d$pop2 == "WEST" & is.na(d$replicate)] <- "i100"
d$replicate[d$pop2 == "ECAN" & is.na(d$replicate)] <- "i100"

# add group-replicate variable for plotting
# get mean per generation

d <- d %>% 

  unite("grouprep", pop2:replicate, remove=FALSE) %>%
  mutate(pop2=factor(pop2, levels=c("ECAN", "RM6", "WAC", "ORC7", "LAS"))) %>%
  # calibrate for different average age of samples
  mutate(Gen2 = case_when(pop2=="LAS" ~ Generation + (20/2) ,
                          pop2=="ORC7" ~ Generation + (2/2),
                          pop2=="RM6" ~ Generation + (6/2) ,
                          pop2=="WAC" ~ Generation + (10/2),
                          pop2 == "ECAN" ~ Generation + (6/2) )) %>%
  # convert from generations to years
  mutate(year=2020-(Gen2*2))
#%>%
#  filter(Geometric_mean > 20) # around here is where GONE seems to be unreliable (populations dive toward this...)

d.mean <- d %>%
  group_by(pop2, year) %>%
  summarize(mean=mean(Geometric_mean)) %>%
  filter(year > 1650) %>%
  arrange(pop2, desc(year))



col_mypop <- mycoldf %>%
  filter(pop2 %in% c("RM", "WAC", "ORC", "LAS", "EAST")) %>%
  filter(pop2 != "WAC") %>%
  pull(mycol)

gone.plot.W <- ggplot() +
  geom_line(data=d %>%   # remove WAC
              filter(pop2 != "WAC"), aes(x=year, y=Geometric_mean, color=pop2, group=grouprep), size=0.25, alpha=0.3) +
  geom_line(data=d.mean %>%   # remove WAC
              filter(pop2 != "WAC"), aes(col=pop2, x=year, y=mean), size=1.5) +
  scale_color_manual(values=col_mypop, labels=c( "ECAN", "RM",  "ORC", "LAS")) +
  scale_x_continuous()  +
  scale_x_reverse(breaks=rev(seq(1000,2000,50)), labels=rev(seq(1000,2000,50)), expand=c(0,0)) +
  scale_y_continuous(labels = function(x) format(x, scientific = FALSE)) +
  #scale_y_continuous(trans = log10_trans(),
  #                   breaks = trans_breaks("log10", function(x) 10^x),
  #                   labels = trans_format("log10", math_format(10^.x))) +
  coord_cartesian(xlim=c(2001,1665),ylim=c(0,150000)) +
  #annotation_logticks(sides="lr") +
  labs(x="Year", y="Ne", color="Population") +
  theme_light() +
  theme(
    legend.position = c( 0.1, 0.3),
    panel.grid = element_blank(),
    axis.text=element_text(size=10),
    axis.title=element_text(size=10)
  )
gone.plot.W

d.mean.zoom <- d.mean

gone.plot.W.zoom <- ggplot() +
  geom_line(data=d %>%   # remove WAC
              filter(pop2 != "WAC"), aes(x=year, y=Geometric_mean, color=pop2, group=grouprep), size=0.25, alpha=0.3) +
  geom_line(data=d.mean.zoom %>%   # remove WAC
              filter(pop2 != "WAC"), aes(col=pop2, x=year, y=mean), size=1.5) +
  scale_color_manual(values=col_mypop) +
  scale_x_continuous(breaks=seq(0,3000,50), labels=seq(0,3000,50), expand=c(0,0)) +
  scale_y_continuous(labels = function(x) format(x, scientific = FALSE), breaks=seq(0,5000,1000)) +
  coord_cartesian(xlim=c(2001,1825), ylim=c(0,2550)) +
  #annotation_logticks(sides="lr") +
  labs(x="", y="", color="Population") +
  theme_light() +
  theme(
    legend.position = "none",
    panel.grid = element_blank()
  )
gone.plot.W.zoom


# plot inset
gone.plot.W.winset <- gone.plot.W %>% 
  ggdraw() +
  draw_plot(
    {
      gone.plot.W.zoom
    },
    x = 0.11, 
    y = 0.6,
    width = 0.52, 
    height = 0.333)

gone.plot.W.winset

save_plot("C:/Users/Cate/Dropbox/WGSfoxes/writing/revision1_MBE/gone/gone_combined.png", gone.plot.W.winset, base_width=6, base_height=3.5)
save_plot("C:/Users/Cate/Dropbox/WGSfoxes/writing/revision1_MBE/gone/gone_combined.pdf", gone.plot.W.winset, base_width=6, base_height=3.5)

```
